flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/PushPartitionIntoTableSourceScanRule.scala

```diff
@@ -51,7 +52,7 @@ class PushPartitionIntoTableSourceScanRule extends RelOptRule(
     scan.getTable.unwrap(classOf[TableSourceTable[_]]) match {
       case table: TableSourceTable[_] =>
         table.tableSource match {
-          case p: PartitionableTableSource => p.getPartitionFieldNames.nonEmpty
+          case _: PartitionableTableSource => true
```

Comment:
see comment on `BatchExecSink`, we should rely on `CatalogTable.isPartitioned` instead of `PartitionableTableSource`?

Reply:
Yes.